### DISCUSSION TITLE
UI Automation

### DISCUSSION THREAD

**Discussion by liorp** [danielpaulus/go-ios#200] [Q&A]
Hi!
Can this library be used for hi automation, such as generating taps and swipes, on a real device?

**Comment by danielpaulus** [comment]
yes. That is the reason it exists :-)  You need to do a couple of things first though. Mainly, installing WebDriverAgent.

**Reply by liorp** [reply]
Neat! Can you reference it in the FAQ? I couldn’t find an appropriate command. What I mean is a command like `ios tap` that would work system wide, device and simulators.

**Reply by danielpaulus** [reply]
So what you need to do is get a paid apple developer account, sign and build webdriveragent, install it and then use a couple of http calls to do this. 
You can also use XCode and install webdriver agent with a free developer account without paying the 100$ every year, but then the installation will only be valid for a week until you have rebuild WDA and install it again on the device. 
I got all the tools in place to automate this really easily. Let me know which option you prefer and I will create a blog post.

**Reply by liorp** [reply]
I think this is the reason I really love the internet. It would be super cool if you had a guide like that 😻 I’m trying to automate some super basic stuff (using Python) and both options seem like a really good fit- but I wasn’t so sure of the technical bits like the installation and the system protections. Thank you so much kind person!  🙏
